analyzers/src/SonarAnalyzer.CSharp/Rules/PropertyToAutoProperty.cs

```diff
@@ -94,27 +90,21 @@ private static bool HasDifferentModifiers(SyntaxList<AccessorDeclarationSyntax>
             return accessors.Skip(1).Any(a => !modifiers.SetEquals(GetModifierKinds(a)));
         }
 
-        private static IEnumerable<SyntaxKind> GetModifierKinds(AccessorDeclarationSyntax accessor)
-        {
-            return accessor.Modifiers.Select(m => m.Kind());
-        }
+        private static IEnumerable<SyntaxKind> GetModifierKinds(AccessorDeclarationSyntax accessor) =>
+            accessor.Modifiers.Select(m => m.Kind());
 
         private static bool TryGetFieldFromSetter(AccessorDeclarationSyntax setter, SemanticModel semanticModel, out IFieldSymbol setterField)
         {
             setterField = null;
 
-            var assignment = GetAssignmentFromBody(setter.Body)
-                ?? GetAssignmentFromExpressionBody(setter.ExpressionBody());
+            var assignment = GetAssignmentFromBody(setter.Body) ?? GetAssignmentFromExpressionBody(setter.ExpressionBody());
 
-            if (assignment != null &&
-                assignment.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
-                assignment.Right != null &&
-                semanticModel.GetSymbolInfo(assignment.Right).Symbol is IParameterSymbol parameter &&
-                parameter.Name == "value" &&
-                parameter.IsImplicitlyDeclared)
+            if (assignment != null
+                && assignment.IsKind(SyntaxKind.SimpleAssignmentExpression)
+                && assignment.Right != null
+                && semanticModel.GetSymbolInfo(assignment.Right).Symbol is IParameterSymbol {Name: "value", IsImplicitlyDeclared: true})
             {
-                return TryGetField(assignment.Left, semanticModel.GetDeclaredSymbol(setter).ContainingType,
-                    semanticModel, out setterField);
+                return TryGetField(assignment.Left, semanticModel.GetDeclaredSymbol(setter).ContainingType, semanticModel, out setterField);
```

Comment:
_Line 112 below:_ There's `&&` refactoring refugee. That condition could be single line.